File: Sprinboot/galeiraibq/src/main/java/pelayo/proyecto/galeiraibq/model/Autor.java
package pelayo.proyecto.galeiraibq.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.SQLRestriction;

import java.io.Serializable;
import java.util.List;

@Entity
@SQLRestriction("estado_borrado = false")
@Data
@Table(name = "autor")
public class Autor implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nombre;
    private String apellidos;
    private Integer fecha_nacimiento;
    private Integer fecha_muerte;
    private String corriente_artistica;
    private String lugar_nacimiento;
    private Boolean estado_borrado;

    @OneToMany(mappedBy = "autor")
    private List<AutorTecnica> autorTecnicas;
}

File: Angular/app-museo-ibq/src/app/shared/models/autor.ts
export interface Autor {
  id: number;
  nombre: string;
  apellidos: string;
  fecha_nacimiento: number | null;
  fecha_muerte: number | null;
  corriente_artistica: string | null;
  lugar_nacimiento: string | null;
  estado_borrado?: boolean;
}

export interface AutorRequest {
  nombre: string;
  apellidos: string;
  fecha_nacimiento: number | null;
  fecha_muerte: number | null;
  corriente_artistica: string | null;
  lugar_nacimiento: string | null;
}
